pkgdb/src/registry/floxpkgs.cc

```diff
@@ -0,0 +1,131 @@
+/* ========================================================================== *
+ *
+ * @file registry/floxpkgs.cc
+ *
+ * @brief Provides a specialized `FloxFlake' which applies rules/pre-processing
+ *        to a `flake' before it is evaluated.
+ *        This is used to implement the `floxpkgs' catalog.
+ *
+ *
+ * -------------------------------------------------------------------------- */
+
+#include <fstream>
+#include <iostream>
+#include <optional>
+
+#include "flox/core/util.hh"
+#include "flox/registry/floxpkgs.hh"
+
+
+/* -------------------------------------------------------------------------- */
+
+namespace flox {
+
+/* -------------------------------------------------------------------------- */
+
+#ifndef RULES_JSON
+#  error "RULES_JSON must be defined"
+#endif  // ifndef RULES_JSON
+
+
+/* -------------------------------------------------------------------------- */
+
+[[nodiscard]] static const std::filesystem::path
+getRulesFile()
+{
+  return nix::getEnv( "_PKGDB_NIXPKGS_RULES_JSON" ).value_or( RULES_JSON );
```

Comment:
do you intend to add a --rules flag to scrape?
defining by env seems a bit weird as an interface where everything else is passed by flag

Reply:
No there will not be a flag until we support custom catalogs.

I had added a `registry` arg as a part of an interface in the past just for us to start hard coding it and needing to invent `--ga-registry`.

I'm trying to avoid similar wasted effort this time by saying - "you must gard code the rules in the binary at build time".

Similarly to `--ga-registry` I created an emv var for use in the test suite but this is not user facing and Rust should not rely on it outside of unit test applications.